addons/mail/static/src/models/composer/composer.js

```diff
@@ -81,6 +82,47 @@ function factory(dependencies) {
             });
         }
 
+        /**
+         * Open the full composer modal.
+         */
+        async openFullComposer() {
+            const attachmentIds = this.attachments.map(attachment => attachment.res_id);
+
+            const context = {
+                default_attachment_ids: attachmentIds,
+                default_body: mailUtils.escapeAndCompactTextContent(this.textInputContent),
+                /* FIXME would need to use suggested_partners here task-2280157 */
+                // default_partner_ids: partnerIds,
+                default_is_log: this.isLog,
+                mail_post_autofollow: true,
+            };
+
+            if (this.thread) {
+                context.default_model = this.thread.model;
+                context.default_res_id = this.thread.id;
+            }
```

Comment:
The guarding is weird. `mail.composer` is modeled as a part of `mail.thread`, so functionally a `mail.composer` doesn't exist without `mail.thread` (minus accidental-and-critical-and-to-be-fixed time frames like teardown in tests, but this code is not called during this time).